cmd/controller/apps_reconciler.go

```diff
@@ -35,5 +43,53 @@ func (r *AppsReconciler) Reconcile(request reconcile.Request) (reconcile.Result,
 		return reconcile.Result{}, err
 	}
 
-	return r.appFactory.NewCRDApp(existingApp, log).Reconcile()
+	force := false
+	crdApp := r.appFactory.NewCRDApp(existingApp, log)
+	if !r.areAppRefsUpToDate(crdApp.GetSecretRefs(), secret, existingApp) ||
+		!r.areAppRefsUpToDate(crdApp.GetConfigMapRefs(), configmap, existingApp) {
+		force = true
+	}
+
+	result, err := crdApp.Reconcile(force)
+	if err != nil {
+		return result, err
+	}
+
+	// Only update if reconcile is successful.
+	// Leave as not updated if error occurs
+	// in case follow up reconcile request may
+	// address issue for App.
+	r.appRefTracker.MarkAppUpdated(existingApp.Name, existingApp.Namespace)
```

Comment:
dont think we need this. if things have failed then app will be re-enqueued because of that and reconcile will try to reconcile as well.

Reply:
Removed and now always marking app has been updated.